pygeo/parameterization/DVGeoSketch.py

```diff
@@ -0,0 +1,296 @@
+       perform embarrassingly parallel finite differencing. Defaults to
+       MPI.COMM_WORLD.
+
+    scale : float
+       A global scale factor from the ESP/VSP geometry to incoming (CFD) mesh
+       geometry. For example, if the ESP/VSP model is in inches, and the CFD
+       in meters, scale=0.0254.
+
+    """
+
+    def __init__(self, fileName, comm=MPI.COMM_WORLD, scale=1.0, projTol=0.01):
+        super().__init__(fileName=fileName)
+
+        # this scales coordinates from model to mesh geometry
+        self.modelScale = scale
+        # and this scales coordinates from mesh to model geometry
+        self.meshScale = 1.0 / scale
+        self.projTol = projTol * self.meshScale  # default input is in meters.
+
+        self.updatedJac = {}
+        self.comm = comm
+
+        # Initial list of DVs
+        self.DVs = OrderedDict()
+
+    @abstractmethod
+    def addPointSet(self, points, ptName, **kwargs):
+        """
+        Add a set of coordinates to DVGeometry
+        The is the main way that geometry, in the form of a coordinate
+        list is given to DVGeometry to be manipulated.
+        """
+        pass
+
+    @abstractmethod
+    def setDesignVars(self, dvDict):
+        """
+        Standard routine for setting design variables from a design
+        variable dictionary.
+
+        Parameters
+        ----------
+        dvDict : dict
+            Dictionary of design variables. The keys of the dictionary
+            must correspond to the design variable names. Any
+            additional keys in the dv-dictionary are simply ignored.
+        """
+        pass
+
+    def getValues(self):
+        """
+        Generic routine to return the current set of design
+        variables. Values are returned in a dictionary format
+        that would be suitable for a subsequent call to setValues()
+
+        Returns
+        -------
+        dvDict : dict
+            Dictionary of design variables
+        """
+        dvDict = OrderedDict()
+        for dvName in self.DVs:
+            dvDict[dvName] = self.DVs[dvName].value
+
+        return dvDict
+
+    @abstractmethod
+    def update(self, ptSetName, config=None):
+        pass
+
+    def pointSetUpToDate(self, ptSetName):
+        """
+        This is used externally to query if the object needs to update
+        its pointset or not. Essentially what happens, is when
+        update() is called with a point set, the self.updated dict
+        entry for pointSet is flagged as true. Here we just return
+        that flag. When design variables are set, we then reset all
+        the flags to False since, when DVs are set, nothing (in
+        general) will up to date anymore.
+
+        Parameters
+        ----------
+        ptSetName : str
+            The name of the pointset to check.
+        """
+        if ptSetName in self.updated:
+            return self.updated[ptSetName]
+        else:
+            return True
+
+    @abstractmethod
+    def getNDV(self):
```

Comment:
I tried that but the method in VSP returns the length of a different list than ESP does so using the same for both broke ESP tests so I just kept them separate instead of figuring out why they're different.